go/api/config.go

```diff
@@ -104,6 +124,18 @@ func (config *baseClientConfiguration) toProtobuf() *protobuf.ConnectionRequest
 		request.ClientName = config.clientName
 	}
 
+	if config.clientAZ != "" {
+		request.ClientAz = config.clientAZ
+	}
+
+	if request.ReadFrom == protobuf.ReadFrom_AZAffinity ||
+		request.ReadFrom == protobuf.ReadFrom_AZAffinityReplicasAndPrimary {
+		if config.clientAZ == "" {
+			panic(
```

Comment:
don't panic, 42

Update `toProtobuf` to return a tuple with error. It is a private API, safe to change.